Fixing several issues

ViaExplore opened the pull request:
Hi @cederom ,

here are the separated commits for reported issues. The last commit is about to discuss a little bit. See comments in code at last commit :)

Regards,
Tomas Kamenicky

ViaExplore committed:
Adding stuff to .gitignore because of Eclipse SDK

ViaExplore committed:
Enhancement: fix the flush()

Created new fucntion libswd_log_flush() which will log and then flush
the stdout. Function is used in libswd_memap.c for speed logging.

ViaExplore committed:
bug fix the =+

Fixed all =+ with +=.

ViaExplore committed:
Bug fix obey cmdq queue length limit

Createt a new struct libswd_stats_t for cmdqlen which is used for memory
limitation management in function libswd_cmd_enqueue(...). CMD queue len
is configured by maxcmdqlen and all libswd_cmdq_... functions handles
correctly cmdqlen. It was necessary for function parameter changes.
Solution with counting directly from queue chain is unacceptable for
embedded platform.

Found a bug at libswd_deinit(...), there is no memory free for
libswdctx->driver.

Maybe an improvement for libswd_cmdq_append(...) which could make a
queue startpoint automatically. Let the autor decide.

cederom commented:
Yea, this is it! Each functional update in a separate commit :-)

Why do you define `libswd_log_flush()` in two places?
What is the reason to use `__attribute__ ((weak))`?

ViaExplore commented:
Why do you define libswd_log_flush() in two places?
- otherwise compilation will fail on libswd_log_flush not declared. Do not ask me why is this :D
BTW libswd_log is also twice, and it wasnt me ;-)

What is the reason to use __attribute__ ((weak))?
- the reason is very simple, if you not define any your custom application function, than the weak one will still work for you directly form library. I think that this is the correct way of doing application interfaces. Or am I wrong?

ViaExplore committed:
Added new CPU revision support

- Added new CPUID for ARM Cortex-M3 r2p0 0x412FC230
- Added lines for storing CPUID init results into
libswdctx->log.debug.cpuid...

ViaExplore committed:
Added new CPU revision support

- Added new CPUID for ARM Cortex-M7

ViaExplore committed:
Fixed memory limition issue with SWD error handler

- libswd_error.c is not counting with memory limitation by stats, thus
implemented a minimum handling.

TODO: libswd_error.c should be revised and repaired, see TODO comments.
